sale_stock_picking_note/models/stock_picking.py

```diff
@@ -1,10 +1,23 @@
 # Copyright 2021 Tecnativa - Víctor Martínez
 # License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl).
 
-from odoo import fields, models
+from odoo import api, fields, models
 
 
 class StockPicking(models.Model):
     _inherit = "stock.picking"
 
     customer_note = fields.Text(string="Customer Comments")
+
+    @api.onchange("partner_id", "picking_type_id")
+    def _onchange_partner_id_picking_notes(self):
+        for picking in self:
```

Comment:
suggestion: you don't need to iterate over the recordset in an onchange

Reply:
You are right